vllm/multimodal/utils.py

```diff
@@ -0,0 +1,61 @@
+import base64
+from io import BytesIO
+from typing import Union
+
+import requests
+from PIL import Image
+
+from vllm.config import ModelConfig
+from vllm.envs import VLLM_IMAGE_FETCH_TIMEOUT
+from vllm.multimodal.image import ImagePixelData
+from vllm.utils import make_async
+
+
+def load_image_from_base64(image: Union[bytes, str]) -> Image.Image:
+    """Load image from base64 format"""
+    return Image.open(BytesIO(base64.b64decode(image)))
+
+
+def fetch_image(image_url: str) -> Image.Image:
+    """Load image from a url or base64 encoded openai GPT4V format"""
+
+    # Avoid circular import
+    from vllm import __version__ as VLLM_VERSION
+
+    headers = {"User-Agent": f"vLLM/{VLLM_VERSION}"}
+    if image_url.startswith('http'):
+        response = requests.get(image_url,
+                                headers=headers,
+                                timeout=VLLM_IMAGE_FETCH_TIMEOUT)
+        response.raise_for_status()
+
+        # Open the image using PIL
+        img = Image.open(BytesIO(response.content))
```

Comment:
Is this more efficient than the streaming approach used in #4200?

Reply:
I believe this is the proper way to read an image data (albeit I haven't done any testing). See https://stackoverflow.com/questions/7391945/how-do-i-read-image-data-from-a-url-in-python

Since we're going to `await` this anyways, would streaming vs non-streaming matter here?

Reply:
Based on [this requests documentation](https://requests.readthedocs.io/en/latest/user/quickstart/#binary-response-content), it seems you're right. Let's keep this code then.